doc/CONFIGURATION.md

```diff
@@ -141,3 +149,453 @@ node:
 # the file we will dump logs into
 log_file: ./logger.log
 ```
+
+# Datastore Configuration
+
+The `datastore` section(s) of the yaml config file is used to config key-value datastores. These key-value datastores are used throughout the entire libp2p, and ipfs stack. Every `datastore` section shares the same configuration options with each other. In fact in the code-base, they all use the same code.
+
+One thing thing to note is the section of the `node` configuration called `storage`. This is really just an aliased/renamed `datastore` section that is used for the storing all our data. By this we mean things like files, documents, movies, etc... It is for all intents and purposes our "main storage".  The one exception to this is the example configuration file earlier in the documentation which enables the `countedStore` which is a reference counted blockstore. If this is used in other `datastore` configuration sections it is simply ignored.
+
+The current datastore types are supported:
+
+* pebble
+  * currently experimental, but is generally the fastest datastore next to pure in-memory stores
+  * it is generally okay, however it can cause unexpected errors, and should be used at your own risk.
+  * not quite as expensive on memory as badger
+* badger
+  * not quite as fast as pebble, but still extremely fast and stable.
+  * The badger default was adjusted to use `fileLoadingMode` with `FileIO` to enable use on low-memory devices.
+  * If using on server environments, change `fileLoadingMode` to 2.
+* leveldb
+  * generally the lowest memory requirements, and suitable for low-memory devices.
+  * generally the slowest datastore
+* memory
+  * consumes the most memory out of everything since it resides entirely in-memory
+  * all data is lost on crash, or restart of service and should only be used for testing
+
```

Comment:
list countedStore here too

Reply:
countedStore shouldn't be listed here, since it can't be used as a stand-alone datastore